Revise remainder of Tutorial 1 for use with LAMMPS-GUI (#19)
lammpstutorials/lammpstutorials-article: lammps-tutorials.tex

at the bottom left of the \textit{Editor} window status bar.  For
running LAMMPS in LAMMPS--GUI it is not required to save the buffer.
The current contents of the buffer will be passed on to LAMMPS.

> However, if you prefer to have you editor buffer automatically saved
> before launching a run or exiting LAMMPS-GUI, there is a setting in the
> \textit{Preferences} dialog for it.

Reviewer: ```suggestion
However, if you prefer to have your editor buffer automatically saved
before launching a run or exiting the LAMMPS-GUI, there is a setting for this
in the \textit{Preferences} dialog.
```

regions for placing the atoms:
{\normalsize
\begin{verbatim}
 # 2) System definition
 region simulation_box block &
     -20 20 -20 20 -20 20
 create_box 2 simulation_box
 region cylinder_in cylinder z 0 0 10 &
     INF INF side in
 region cylinder_out cylinder z 0 0 10 &
     INF INF side out
 create_atoms 1 random 1000 341341 cylinder_out
 create_atoms 2 random 150 127569 cylinder_in
\end{verbatim}
}
The \textit{side in} and \textit{side out} keywords are used to define regions
that are respectively inside and outside of the cylinder of radius 10. Then,
copy similar lines as previously into \textit{input.min.lmp}:
{\normalsize
\begin{verbatim}
 # 3) Settings
 mass 1 1
 mass 2 1
 pair_style lj/cut 2.5
 pair_coeff 1 1 1.0 1.0
 pair_coeff 2 2 0.5 3.0

 # 4) Visualization
 thermo 10
 thermo_style custom step etotal press
 dump mydmp all image 100 dump.min.*.ppn &
    type type shiny 0.1 box no 0.01 &
    view 0 0 zoom 1.8
 dump_modify mydmp acolor 1 darkturquoise &
    acolor 2 darkblue adiam 1 1 adiam 2 3 &
    backcolor white

 # 5) Run
 minimize 1.0e-4 1.0e-6 1000 10000
 write_data minimized_coordinates.data
\end{verbatim}
}

> The main novelty of this input compared to the previous input script,

Reviewer: ```suggestion
The main novelty of this input compared to the previous input script
```

    intersect group_type_2 group_region_out
 delete_atoms group group_type_1_in
 delete_atoms group group_type_2_out
 # we don't need these four groups anymore
 group group_region_in delete
 group group_region_out delete
 group group_type_1_in delete
 group group_type_2_out delete
\end{verbatim}
}

The two first \textit{region} commands recreate the previously defined
regions, which is necessary since regions are not saved by the
\textit{write\_data} command.  The first two \textit{group} commands are
used to create groups containing all the atoms of type 1 and all the
atoms of type 2, respectively.  The next two \textit{group} commands
create atom groups based on their positions at the beginning of the
simulation, i.e. when the commands are being read by LAMMPS.  The last
two \textit{group} commands create atom groups based on the intersection
between the previously defined groups.  Finally, the two
\textit{delete\_atoms} commands delete the atoms of type 1 that are
located within the cylinder and the atoms of type 2 that are located
outside the cylinder, respectively.  LAMMPS has a limited number of
custom groups (30), so it is good practice to delete groups that are

> no longer needed, which is done with the last four lines.

Reviewer: ```suggestion
no longer needed. This is done in the last four lines.
```

\label{fig:mixing}
\end{figure}

There are a few more differences from the previous simulation.  First,
the \textit{velocity create} command assigns an initial velocity to each
atom.  The initial velocity is chosen so that the average initial
temperature is equal to 1 temperature unit.  The additional keywords
ensure that no linear momentum (\textit{mom yes}) and no angular
momentum (\textit{rot yes}) are given to the system and that the
generated velocities are distributed as a Gaussian.  Another improvement
is the \textit{zero yes} keyword in the Langevin thermostat, which
ensures that the total random force applied to the atoms is equal to
zero.

The steps to have no initial linear momentum and no net random force are

> important to avoid that the system will start to drift or move as a

Reviewer: ```suggestion
important to prevent the system from starting to drift or move as a
```

whole.  For a bulk system with periodic boundary conditions, it is

> expected to stay in place and thus when computing the temperature from
> the kinetic energy, we use $3N-3$ degrees of freedom since there is no

Reviewer: ```suggestion
expected to stay in place. Thus, when computing the temperature from
the kinetic energy, we use $3N-3$ degrees of freedom since there is no
```

global translation.  In a drifting system, some of the kinetic energy is

> due to the drift, which means the system itself becomes cooler.  In the
> extreme case it can freeze and drift very fast and thus people sometimes
> refer to this as the ``flying ice cube syndrome''.

Reviewer: ```suggestion
due to the drift, which means the system itself cools down.  In the
extreme case it can freeze and drift very fast. This phenomenon is sometimes referred to as the ``flying ice cube syndrome'' \cite{wong2016good}.
```
